bugshaker/src/main/java/com/github/stkent/bugshaker/MainActivity.java

```diff
@@ -0,0 +1,232 @@
+
+	private DrawingView drawView;
+
+	private ImageButton currPaint;
+
+	@Override
+	protected void onCreate(Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_main);
+
+		drawView = (DrawingView)findViewById(R.id.drawing);
+		LinearLayout paintLayout = (LinearLayout)findViewById(R.id.paint_colors);
+		currPaint = (ImageButton)paintLayout.getChildAt(0);
+		currPaint.setImageDrawable(getResources().getDrawable(R.drawable.paint_pressed));
+
+		ImageButton drawBtn = (ImageButton) findViewById(R.id.draw_btn);
+		drawBtn.setOnClickListener(this);
+
+		drawView.setBrushSize(getResources().getInteger(R.integer.small_size));
+
+		ImageButton eraseBtn = (ImageButton) findViewById(R.id.erase_btn);
+		eraseBtn.setOnClickListener(this);
+
+		ImageButton sendBtn = (ImageButton) findViewById(R.id.sendEmail);
+		sendBtn.setOnClickListener(this);
+
+		ImageButton speechBtn = (ImageButton) findViewById(R.id.speechBox);
+		speechBtn.setOnClickListener(this);
+
+
+		Logger logger = new Logger(true);
+		Application application = getApplication();
+
+		String pathOfScreenshot = getIntent().getStringExtra("uri");
+		File screenshotFile = new File(pathOfScreenshot);
+
+		if(screenshotFile.exists()) {
+			Bitmap myBitmap = BitmapFactory.decodeFile(pathOfScreenshot);
+
+			drawView = (DrawingView) findViewById(R.id.drawing);
+			Drawable temp = new BitmapDrawable(getResources(), myBitmap);
+			drawView.setBackgroundDrawable(temp);
+		}
+		else {
+			Log.d("MainActivity", "File does not exist");
+		}
+	}
+
+	public void setButtonClickListener(String string, ImageButton button, final Dialog dialog, boolean isMarker) {
+		String str = string;
+		ImageButton imageButton = button;
+		final float size;
+
+		switch (str) {
+		case "smallBtn":
+			size = getResources().getInteger(R.integer.small_size);
+			break;
+		case "mediumBtn":
+			size = getResources().getInteger(R.integer.medium_size);
+			break;
+		case "largeBtn":
+			size = getResources().getInteger(R.integer.large_size);
+			break;
+		default:
+			size = 1;
+			break;
+		}
+
+		if (isMarker) {
+			imageButton.setOnClickListener(new OnClickListener() {
+				@Override
+				public void onClick(View view) {
+					drawView.setBrushSize(size);
+					drawView.setLastBrushSize(size);
+					drawView.setErase(false);
+					dialog.dismiss();
+				}
+			});
+		}
+		else {
+			imageButton.setOnClickListener(new OnClickListener() {
+				@Override
+				public void onClick(View view) {
+					drawView.setErase(true);
+					drawView.setBrushSize(size);
+					dialog.dismiss();
+				}
+			});
+		}
+	}
+
+	@Override
+	public void onClick(View view){
+		final Dialog brushDialog = new Dialog(this);
+		brushDialog.setTitle(getString(R.string.brush_dialog_title));
+		brushDialog.setContentView(R.layout.brush_chooser);
+
+		if(view.getId()==R.id.draw_btn){
+
+			ImageButton smallBtn = (ImageButton) brushDialog.findViewById(R.id.small_brush);
```

Comment:
we should extract this outside the if-else-if we have here so that we don't do this twice.